File: bamlcode/python_demo/hello_baml.py
#!/usr/bin/env -S uv run --script
# /// script
# requires-python = ">=3.10"
# dependencies = ["baml-core", "pydantic"]
# ///
"""Tiny Python <-> BAML interop demo.

Calls the BAML function `demo.greet` (baml_src/ns_demo/demo.baml) through the
generated `baml_sdk` package. Greeting comes back as a typed pydantic model.

Regenerate the SDK after editing the BAML side:  baml generate
Run:  uv run python_demo/hello_baml.py   (or just ./python_demo/hello_baml.py)
"""

from baml_sdk import demo  # generated next to this script (baml generate)


def main() -> None:
    for name in ["vbv", "  sheep council  ", "BAML"]:
        greeting = demo.greet(name=name)  # -> Greeting(message=..., letters=...)
        print(f"{greeting.message:30}  ({greeting.letters} letters)")

    # It's a real pydantic model — serialization comes for free.
    print("\nas JSON:", demo.greet(name="python").model_dump_json())

    # BAML class methods are callable from Python too:
    g = demo.Greeting.new("class methods")  # factory (no self) -> staticmethod
    print("\nfactory:  ", g.message)
    print("instance: ", g.shout())  # `shout(self)` -> bound method
    print("chained:  ", g.repeated(3).model_dump())  # methods returning Greeting chain


if __name__ == "__main__":
    main()

File: bamlcode/typescript_demo/hello_baml.ts
// Tiny TypeScript <-> BAML interop demo (the Node twin of python_demo/hello_baml.py).
//
// Calls the BAML function `demo.greet` (baml_src/ns_demo/demo.baml) through the
// generated `baml_sdk` package. The greeting comes back as a typed class.
//
// Regenerate the SDK after editing the BAML side:  baml generate
// Run:  pnpm hello   (from typescript_demo/, after `pnpm install`)
import { demo } from "./baml_sdk/index.js"; // generated next to this script (baml generate)

function main(): void {
  for (const name of ["vbv", "  sheep council  ", "BAML"]) {
    const greeting = demo.greet({ name }); // -> Greeting { message, letters }
    console.log(`${greeting.message.padEnd(30)}  (${greeting.letters} letters)`);
  }

  // It's a plain typed object — JSON serialization comes for free (the bound
  // method fields are functions, so JSON.stringify drops them).
  console.log("\nas JSON:", JSON.stringify(demo.greet({ name: "python" })));

  // BAML class methods are callable from TypeScript too:
  const g = demo.Greeting.new("class methods"); // factory (no self) -> static method
  console.log("\nfactory:  ", g.message);
  console.log("instance: ", g.shout()); // shout(self) -> bound instance method
  // methods returning Greeting chain; strip the bound methods for a clean dump.
  console.log("chained:  ", JSON.parse(JSON.stringify(g.repeated(3))));
}

main();
